telegram/__init__.py

```diff
@@ -160,40 +172,109 @@ def update_item(self, item, caller=None, source=None, dest=None):
+    
+    # def _photo_url_broadcast(self, url, msg):
+        # for cid in self._chat_ids:
+            # try:
+                # self._bot.sendPhoto(chat_id=cid, photo=url)
+            # except Exception as e:
+                # self.logger.error("could not broadcast to chat id [%d] %s %s error %s" % (cid, url, msg, e))
 
     def message_handler(self, msg):
         self._chat_id = msg['chat']['id']
         tmp_chat_id = msg['chat']['id']
-        command = msg['text']
-
-        self.logger.info("[%d] command received: %s" % (self._chat_id, command))
+        msg_list = msg['text'].split(' ')
+        para = ""
+        if len(msg_list) > 1:
+            command = msg_list[0]
+            para = msg_list[1]
+            self.logger.info("[%d] command received: %s para: %s" % (self._chat_id, command, para))
+        else:
+            command = msg['text']
+            self.logger.info("[%d] command received: %s" % (self._chat_id, command))
 
         # /roll: just a dummy to test interaction
         if command == '/roll':
-                self._bot.sendMessage(self._chat_id, random.randint(1,6))
+            self._bot.sendMessage(self._chat_id, random.randint(1,6))
 
-	# /time: return server time
+        # /time: return server time
         elif command == '/time':
-                self._bot.sendMessage(self._chat_id, str(datetime.datetime.now()))
+            self._bot.sendMessage(self._chat_id, str(datetime.datetime.now()))
 
         # /help: show available commands als keyboard
         elif command == '/help':
-	        self._bot.sendMessage(self._chat_id, "choose", reply_markup={"keyboard":[["/roll","/hide"], ["/time","/list"]]})
+	        self._bot.sendMessage(self._chat_id, "choose", reply_markup={"keyboard":[["/roll","/hide"], ["/time","/list"], ["/lo","/info"]]})
 
         # /hide: hide keyboard
         elif command == '/hide':
-                hide_keyboard = {'hide_keyboard': True}
-                self._bot.sendMessage(self._chat_id, "I'll hide the keyboard", reply_markup=hide_keyboard)
+            hide_keyboard = {'hide_keyboard': True}
+            self._bot.sendMessage(self._chat_id, "I'll hide the keyboard", reply_markup=hide_keyboard)
 
         # /list: show registered items and value
         elif command == '/list':
             self.list_items(self._chat_id)
+            
+        # /info: show item-menu with registered items with specific attribute
+        elif command == '/info':
+            #self._bot.sendMessage(self._chat_id, "Infos from the items:", reply_markup=self.create_info_reply_markup())
+            self._bot.sendMessage(self._chat_id, "Infos from the items:", reply_markup={"keyboard":self.create_info_reply_markup()})
+            
+        # /lo: show logics
+        elif command == '/lo':
+            tmp_msg = "";
+            tmp_msg+="Logics:\n"
+            for logic in sorted(self._sh.return_logics()):
+                data = []
+                lo = self._sh.return_logic(logic)
```

Comment:
The call to `self._sh.return_logics()` can be made forward compatible by using `Logics.return_loaded_logics()` instead.

Reply:
I can not find any more information about this method Logics.return_loaded_logics(). Is this part of SH?

Reply:
Details are in the developer docs: http://smarthomeng.de/developer/lib/logic.html#module-lib.logic

Reply:
from lib.logic import Logics

self.logics = Logics.get_instance()
for logic in sorted(self.logics.return_loaded_logics()):

AttributeError: 'NoneType' object has no attribute 'return_loaded_logics'

I have to do that Download or update the library (lib.logic)

Reply:
How do you run your SHNG instance? Usually you have to checkout the recent SHNG [code](https://github.com/smarthomeNG/smarthome/tree/develop) (in develop branch) and inside the plugins directory the SHNG [plugins](https://github.com/smarthomeNG/plugins/tree/develop) (develop branch too).

When developing plugin enhancement you usually use your own repository checkout of the plugins repo.

When the method is missing it means that you do not have a recent develop checkout of the core, because the [current version](https://github.com/smarthomeNG/smarthome/blob/develop/lib/logic.py#L587) contains this.

Reply:
ok, I use SH-Master 1.4.1  I could have imagined that.
I first have to install the developer version parallel.

Reply:
This is also in the master branch of SHNG - but maybe not in your fork (beacuse it was not updated).

If this all is too complicated we can merge it and refactor this later.

Reply:
at which point in the code I place "self.logics = Logics.get_instance ()". If I do this in the "__init__" function, then the result of .get_instance () is "None"?

Reply:
Because logics are initialized after plugins and items are loaded and initialized, you can't put it in the `__init__` routine. Place it at the start of the run method.

Reply:
Many Thanks. I now have access to logic via the new lib.logic-api. I hope it fits. At least it works for me. Sorry for my "good" english

Reply:
No problem, thanks for your adjustments! Just two notes:
* [line 96](https://github.com/smarthomeNG/plugins/pull/108/commits/14b203678eb77f619149857b6044c2dce894cff5#diff-f2dfe37a3fe68465cab83f206c469e45R96) contains a TAB character for indentation, usually we use 4 spaces (instead of one TAB)
* [line 254](https://github.com/smarthomeNG/plugins/pull/108/commits/14b203678eb77f619149857b6044c2dce894cff5#diff-f2dfe37a3fe68465cab83f206c469e45R254) contains `self._sh.return_logic()` which can also be replaced with `self.logics.return_logic()`